redbot/core/settings_caches.py

```diff
@@ -56,6 +56,54 @@ async def set_prefixes(
             await self._config.guild_from_id(gid).prefix.set(prefixes)
 
 
+class I18nManager:
+    def __init__(self, config: Config):
+        self._config: Config = config
+        self._guild_locale: Dict[int, str] = {}
+        self._guild_regional_format: Dict[int, str] = {}
+
+    async def get_locale(self, guild: Union[discord.Guild, None]) -> str:
+        """Get the guild locale from the cache"""
+        if guild is None:
+            # Return the bot's globally set locale if its None on a guild scope.
+            out = await self._config.locale()
+        elif guild.id in self._guild_locale:
+            out = self._guild_locale[guild.id]
+        else:
+            out = await self._config.guild(guild).locale()
+            if out is None:
+                out = await self._config.locale()
+            else:
+                self._guild_locale[guild.id] = out
+        return out
+
+    async def set_guild_locale(self, guild: discord.Guild, locale: Union[str, None]) -> None:
+        """Set the guild locale in the config and cache"""
+        self._guild_locale[guild.id] = locale
+        await self._config.guild(guild).locale.set(locale)
+
+    async def get_regional_format(self, guild: Union[discord.Guild, None]) -> Optional[str]:
```

Comment:
Same comment as for `get_locale()` method.

But I also have one more thought. I think it would be more logical to use this resolution order for guild regional format:
- use guild regional format, if not None
- use guild locale, if not None
- use global regional format, if not None
- use global locale

It does seem to make some sense to me, but I am not really sure if it's actually better or worse than current resolution order which is:
- use guild regional format, if not None
- use global regional format, if not None
- use guild locale, if not None
- use global locale

// Edit: Hmm, it might require more rewriting, so I'm not sure whether it's really worth pursuing.

Reply:
Theoretically should be done?